### ISSUE TITLE
Godot: add mono build

### ISSUE THREAD

**Issue by nico202** [NixOS/nixpkgs#42160]
## Issue description

Godot supports c# from version 3.0. The current nix pkg builds without it being enabled
link with info: http://docs.godotengine.org/en/3.0/development/compiling/compiling_with_mono.html#doc-compiling-with-mono

@twey

**Comment by piq9117** [comment]
is anybody still interested in this? I got this locally.

**Comment by piq9117** [comment]
@Twey is this something i can work on?

**Comment by lihop** [comment]
I opened a [draft pull request](https://github.com/NixOS/nixpkgs/pull/88006) for a derivation I am successfully using to run and build Godot projects with mono. However, I am still experiencing some weirdness around NuGet. More details in the PR description.

**Comment by ThibaultLemaire** [comment]
I would be interested in this

**Comment by ThibaultLemaire** [comment]
For anyone interested and in the meantime, I am able to run [the official Godot Mono version](https://downloads.tuxfamily.org/godotengine/3.2.3/mono/Godot_v3.2.3-stable_mono_x11_64.zip) on NixOS in a chroot env using the following `shell.nix`:

``` nix
{ pkgs ? import <nixpkgs> {} }:

(pkgs.buildFHSUserEnv {
  name = "godot-mono-env";
  targetPkgs = pkgs: (with pkgs;
    [
      libpulseaudio
      xorg.libX11
      xorg.libXcursor
      xorg.libXinerama
      xorg.libXrandr
      xorg.libXrender
      xorg.libXi
      alsaLib
      libGL
      msbuild
    ]);
}).env
```

**Comment by ln42** [comment]
Just sharing an humble overlay (`~/.config/nixpkgs/overlays/godot-mono/default.nix`) based on @ThibaultLemaire env:
```
self: super: {
  godot-mono = with super;
  let
    arch = "64";
    version = "3.2.3";
    pkg = stdenv.mkDerivation  {
      name = "godot-mono-unwrapped";
      buildInputs = [ unzip ];
      unpackPhase = "unzip $src";
      version = version;
      src = fetchurl {
        url = "https://downloads.tuxfamily.org/godotengine/${version}/mono/Godot_v${version}-stable_mono_x11_${arch}.zip";
        sha256 = "4ad458402da2870203ea5db1653f41e927236fe95465bc4d39ae5bff29e32154";
      };
      installPhase = ''
        cp -r . $out
      '';
    };
  in buildFHSUserEnv {
    name = "godot-mono";
    targetPkgs = pkgs: (with pkgs;
      [ libpulseaudio
        xorg.libX11
        xorg.libXcursor
        xorg.libXinerama
        xorg.libXrandr
        xorg.libXrender
        xorg.libXi
        alsaLib
        libGL
        msbuild
      ]);
    runScript = "${pkg.outPath}/Godot_v${version}-stable_mono_x11_${arch}/Godot_v${version}-stable_mono_x11.${arch}";
  };
}
```

**Comment by dustinlacewell** [comment]
With the above I'm getting:
```
$ godot-mono                                                                                                                               ~
/nix/store/7hqwsi7ky392fa48f7mvbflrphaq8yyr-godot-mono-unwrapped/Godot_v3.3.2-stable_mono_x11_64/Godot_v3.3.2-stable_mono_x11.64: error while loading shared libraries: libXext.so.6: cannot open shared object file: No such file or directory
```

**Comment by lihop** [comment]
@dustinlacewell Here's what I'm using to run Godot 3.3.2 on nixos-unstable, based on @ln42's overlay:
```
self: super: {
  godot-mono = with super;
  let
    arch = "64";
    version = "3.3.2";
    releaseName = "stable";
    subdir = "";
    pkg = stdenv.mkDerivation  {
      name = "godot-mono-unwrapped";
      buildInputs = [ unzip ];
      unpackPhase = "unzip $src";
      version = version;
      src = fetchurl {
        url = "https://downloads.tuxfamily.org/godotengine/${version}${subdir}/mono/Godot_v${version}-${releaseName}_mono_x11_${arch}.zip";
        sha256 = "sha256-imyJ/oYjZ++EeEdN99JrUuBapMr40PSuGWcuRTEc0Qw=";
      };
      installPhase = ''
        cp -r . $out
      '';
    };
  in buildFHSUserEnv {
    name = "godot-mono";
    targetPkgs = pkgs: (with pkgs;
      [ alsaLib
        dotnetCorePackages.sdk_5_0
        libGL
        libpulseaudio
        libudev
        xorg.libX11
        xorg.libXcursor
        xorg.libXext
        xorg.libXi
        xorg.libXinerama
        xorg.libXrandr
        xorg.libXrender
        zlib
      ]);
    runScript = "${pkg.outPath}/Godot_v${version}-${releaseName}_mono_x11_${arch}/Godot_v${version}-${releaseName}_mono_x11.${arch}";
  };
}
```
Also, in Godot under 'Editor' -> 'Editor settings...' -> 'Mono' -> 'Builds', I have 'Build Tool' set to 'dotnet CLI'.

**Comment by tomodachi94** [comment]
Appears to have been resolved in #177578:

https://search.nixos.org/packages?channel=23.11&from=0&size=50&sort=relevance&type=packages&query=godot+mono
